src/api/drone.rs

```diff
@@ -1,5 +1,20 @@
-use crate::types::packet::Packet;
+use crate::types::channel::Channel;
+use crate::types::source_routing_header::NodeId;
+use std::collections::HashMap;
+use std::thread;
 
-pub trait DroneAble {
-    fn forward_packet(&self, packet: Packet) -> bool;
+pub struct Drone {
+    id: NodeId,
+    thread: thread::JoinHandle<()>,
+    channels: HashMap<u8, Channel>,
+}
+
+trait DroneTrait {
+    fn new(drone_id: u8) -> Self;
+    // The thread would be created inside here,
+    // giving him reference to the list of channels,
+    // that would be expanded during the creation of the
+    // later drones.
+    fn add_channel(new_channel: Channel) -> Result<String, String>;
+    fn remove_channel(drone_id: u8) -> Result<String, String>;
```

Comment:
Result<String, String> seem excessive to me. Either Result<(), String> or Result<id/size, String> or bool seem better.

Reply:
Yea it was more a placeholder than anything.